game_of_life.py

```diff
@@ -0,0 +1,122 @@
+import itertools
+import random
+import sys
+import time
+from typing import List
+
+import pygame
+
+from pharmacontroller import SCREEN_SIZE, PharmaScreen
+
+Grid = type(List[List[int]])
+
+
+def is_valid_index(screen: PharmaScreen, row: int, column: int) -> bool:
+    """
+    Check if the given row and column are valid indices.
+
+    Args:
+        row (int): The row index.
+        column (int): The column index.
+
+    Returns:
+        bool: True if the indices are valid, False otherwise.
+    """
+    conditions = (
+        0 <= row < SCREEN_SIZE,
+        0 <= column < SCREEN_SIZE,
+        screen.is_drawable(row, column),  # Todo: transform to static method
+    )
+    return all(conditions)
+
+
+def generate_grid() -> Grid:
+    """
+    Generate a grid without any living cells.
+
+    Returns:
+        numpy.ndarray: The grid state.
+    """
+    return [[0 for _ in range(SCREEN_SIZE)] for _ in range(SCREEN_SIZE)]
+
+
+def get_neighbors(screen: PharmaScreen, grid: Grid, row: int, column: int) -> List[float]:
+    """
+    Get the neighbors of the cell at the given row and column.
+
+    Args:
+        grid (numpy.ndarray): The grid state.
+        row (int): The row of the cell.
+        column (int): The column of the cell.
+
+    Returns:
+        List[int]: The list of neighbors.
```

Comment:
The type hint is `List[float]` but the docstring is `List[int]`

Imo when there are type hints, the docstring does not need to include the type (I know that is optional when using google docstring format). All editors should still show the correct types when editing.